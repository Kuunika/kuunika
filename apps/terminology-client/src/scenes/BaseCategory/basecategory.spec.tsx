import React from 'react';
import { render } from '@testing-library/react';

import { CategoryView } from './index';

describe('Base Category', () => {
  it('should render all categories successfully', () => {
    const data = {
      pageTitle: 'All categories',
      data: [
        {
          title: 'Terms',
          content: 'test',
          onClick: () => {}
        }
      ],
      breadCrumb: ['clinical']
    };
    const { baseElement, getByText, getByTestId, getAllByTestId } = render(
      <CategoryView {...data} />
    );

    expect(getByText(data.pageTitle)).toBeTruthy();
    expect(getByTestId('breadcrumb')).toMatchSnapshot();
    expect(getAllByTestId('category-card').length).toBe(data.data.length);
  });
});
