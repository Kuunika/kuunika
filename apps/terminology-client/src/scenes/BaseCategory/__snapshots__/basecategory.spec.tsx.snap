// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Category should render all categories successfully 1`] = `
<body>
  <div>
    <h1
      class="sc-gzVnrw jDlAHK"
    >
      All categories
    </h1>
    <h2
      class="sc-ifAKCX hKvzgS"
      data-testid="breadcrumb"
    >
      <b>
        Category: 
      </b>
      <span>
        <i
          class="sc-EHOje ihybRz"
        >
          clinical
        </i>
      </span>
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <div
          class="sc-bwzfXH kDFgrS"
          data-testid="category-card"
        >
          <div
            class="sc-htpNat bKNYPk"
          >
            Terms
          </div>
          <p
            class="sc-bxivhb jcWvLS"
          >
            test
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-104 MuiButton-contained"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              VIEW
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
